src/hotspot/share/oops/resolvedIndyEntry.hpp

```diff
@@ -107,20 +107,20 @@ class ResolvedIndyEntry {
   void fill_in(Method* m, u2 num_params, u1 return_type, bool has_appendix) {
     set_num_parameters(num_params);
     _return_type = return_type;
-    set_flags(has_appendix);
+    set_has_appendix(has_appendix);
     // Set the method last since it is read lock free.
     // Resolution is indicated by whether or not the method is set.
     Atomic::release_store(&_method, m);
   }
 
-  // has_appendix is currently the only other flag besides resolution_failed
-  void set_flags(bool has_appendix) {
+  void set_has_appendix(bool has_appendix) {
     u1 new_flags = (has_appendix << has_appendix_shift);
-    assert((new_flags & 1) == 0, "New flags should not change resolution flag");
-    // Preserve the resolution_failed bit
-    _flags = (_flags & 1) | new_flags;
+    u1 old_flags = _flags &  ~(1 << has_appendix_shift);
+    // Preserve the unaffected bits
+    _flags = old_flags | new_flags;
```

Comment:
I may be having a mental blank this late in the week, but why do we need to do anything other than:
```
_flags |= new_flags;
```
? `new_flags` should at most have one bit set (the appendix bit) and OR'ing with zero preserves all other bits.

Reply:
Oooh nah, I just realised. @dholmes-ora , we must have John's version because if we're setting it to `0` and the `_flags` already is `1` then it won't do anything.

Reply:
Hmmm so the complex machinations is not really about "preserving the other bits".

Reply:
I was assuming the only possibility is to go from no-appendix to having-an-appendix

Reply:
Anyway yes we need the original logic, which for the sake of completeness creates `old_flags` such that the appendix bit is zero, and all other bits are preserved, and we then OR with `new_flags` that only has the appendix bit ever set and thuis it determines the final value of the appendix bit.